src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/spotify/PlayerLogic.java

```diff
@@ -0,0 +1,423 @@
+ * 
+ * @author Lars Buttgereit
+ */
+@Service(PlayerLogic.class)
+public class PlayerLogic {
+	private Authorization auth;
+	private String deviceID = null;
+	private Search search;
+	private int volume = 50;
+	// private String deviceName = null;
+	private ArrayList<String[]> actualSearchResult = null;
+	
+	public PlayerLogic() {
+		init();
+	}
+
+	@Init
+	public void init() {
+		this.auth = new Authorization();
+		this.auth.init();
+		setDevice(0);
+		this.search = new Search(this.auth);
+	}
+
+	/**
+	 * needed for the first init. need the clientID and the clientSecret form a
+	 * spotify devloper account
+	 * 
+	 * @param clientID
+	 * @param clientSecret
+	 * @return login link to a personal spotify account
+	 */
+	public URI firstTimeInit(String clientID, String clientSecret) {
+		this.auth = new Authorization();
+		this.auth.setClientID(clientID);
+		this.auth.setClientSecret(clientSecret);
+		return this.auth.authorizationCodeUri();
+	}
+
+	/**
+	 * create the refresh token in he authorization object with the authCode
+	 * 
+	 * @param authCode
+	 *            Callback from the login link
+	 */
+	public void inputAuthCode(String authCode) {
+		this.auth.createRefreshToken(authCode);
+		this.auth.init();
+	}
+
+	/**
+	 * get all devices that logged in at the moment
+	 * 
+	 * @return empty ArrayList if no device available else the name of the devices
+	 */
+	public ArrayList<String> getDevices() {
+
+		ArrayList<String> deviceNames = new ArrayList<>();
+		if (this.auth.getSpotifyApi() != null) {
+			GetUsersAvailableDevicesRequest getUsersAvailableDevicesRequest = this.auth.getSpotifyApi()
+					.getUsersAvailableDevices().build();
+
+			try {
+				Device[] devices = getUsersAvailableDevicesRequest.execute();
+				for (Device device : devices) {
+					deviceNames.add(device.getName());
+				}
+			} catch (SpotifyWebApiException | IOException e) {
+				System.err.println(e);
+				e.printStackTrace();
+			}
+
+			return deviceNames;
+		}
+		System.err.println("please init the authorization object");
+		return deviceNames;
+	}
+
+	/**
+	 * set the given device as acutal active device for playing music
+	 * 
+	 * @param deviceNumber
+	 *            index of the device array. Order is the same as in the output in
+	 *            getDevices
+	 * @return selected device
+	 */
+	public String setDevice(int deviceNumber) {
+		if (this.auth.getSpotifyApi() != null) {
+			GetUsersAvailableDevicesRequest getUsersAvailableDevicesRequest = this.auth.getSpotifyApi()
+					.getUsersAvailableDevices().build();
+			try {
+				Device[] devices = getUsersAvailableDevicesRequest.execute();
+				if (deviceNumber < devices.length) {
+					this.deviceID = devices[deviceNumber].getId();
+					return devices[deviceNumber].getName();
+				}
+				return "This device was not found";
+			} catch (SpotifyWebApiException | IOException e) {
+				e.printStackTrace();
+				System.err.println(e);
```

Comment:
`e.printStackTrace();` prints the whole exception so `System.err.println(e);` is not needed